controllers/registerController.js

```diff
@@ -0,0 +1,24 @@
+import bcrypt from 'bcrypt';
+import { RegisterRepository } from "../repositories/registerRepository.js";
+
+export class RegisterController {
+    constructor() {
+        this.registerRepository = new RegisterRepository();
+    }
+
+    registerUser = async (req, res, next) => {
+        const { username, password } = req.body;
+
+        const hashedPassword = await bcrypt.hash(password, 10);
+
+        try {
+            await this.registerRepository.addUser(username, hashedPassword);
+
+            const token = this.registerRepository.generateToken(username);
```

Comment:
Well, token generation must be part of the service, not repository. Repository only responsible for the data layer